File: frontend/services/api/authApi.ts
import apiClient from './client';

interface RegisterData {
  email:      string;
  contrasena: string;
  nombre?:    string;
}

interface LoginData {
  email:      string;
  contrasena: string;
}

interface AuthResponse {
  mensaje:  string;
  token:    string;
  usuario: {
    id:     number;
    email:  string;
    nombre: string;
    role:   string;
  };
}

interface MeResponse {
  usuario: {
    id:     number;
    email:  string;
    nombre: string;
    role:   string;
  };
}

export const authApi = {
  registro: (data: RegisterData) =>
    apiClient.post<AuthResponse>('/auth/registro', data),

  login: (data: LoginData) =>
    apiClient.post<AuthResponse>('/auth/login', data),

  me: (token: string) =>
    apiClient.get<MeResponse>('/auth/me', {
      headers: { Authorization: `Bearer ${token}` },
    }),

  forgotPassword: (email: string) =>
    apiClient.post('/auth/forgot-password', { email }),

  resetPassword: (token: string, contrasena: string) =>
    apiClient.post('/auth/reset-password', { token, contrasena }),
};

export default authApi;

File: mobile/lib/features/auth/data/auth_api.dart
import 'package:dio/dio.dart';

class AuthApi {
  final Dio dio;
  AuthApi(this.dio);

  Future<Map<String, dynamic>> login({
    required String usuario,
    required String contrasena,
  }) async {
    final res = await dio.post('/api/auth/login', data: {
      'usuario': usuario,
      'contrasena': contrasena,
    });
    return (res.data as Map).cast<String, dynamic>();
  }

  Future<Map<String, dynamic>> register({
    required String usuario,
    required String contrasena,
  }) async {
    final res = await dio.post('/api/auth/registro', data: {
      'usuario': usuario,
      'contrasena': contrasena,
    });
    return (res.data as Map).cast<String, dynamic>();
  }
}
